Fabric.Authorization.API/Modules/UsersModule.cs

```diff
@@ -284,5 +369,24 @@ public class UsersModule : FabricModule<User>
                 request.SecurableItem = client.TopLevelSecurableItem.Name;
             }
         }
+
+        private async Task<Negotiator> ValidateRolesToAdd(List<RoleApiModel> apiRoles)
+        {
+            if (apiRoles.Count == 0)
+            {
+                return await CreateFailureResponse(
+                    InvalidRoleArrayMessage, HttpStatusCode.BadRequest);
+            }
+
+            if (apiRoles.Any(r => !r.Id.HasValue || r.Id.Value == Guid.Empty || string.IsNullOrEmpty(r.Grain) ||
+                                  string.IsNullOrEmpty(r.SecurableItem)))
+            {

```

Comment:
Can we put this line before the `if` and then use the `messages` variable in the `if` so we're not executing this query twice?

Reply:
Good call. I'll make that change.